dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LdapFailoverTest.java

```diff
@@ -205,6 +201,78 @@ private static void updateLibertyServer() throws Exception {
+        /* restarting the primary server */
+        ds1.getLdapServer().startListening();
+        Thread.sleep(120000L);
+        libertyServer.setMarkToEndOfLog(libertyServer.getMostRecentTraceFile());
+        returnUser = servlet.checkPassword(USER_DN, PWD);
+        assertNotNull("Should find user on checkPassword using " + USER_DN, returnUser);
+        List<String> trMsgs = libertyServer.findStringsInLogsAndTraceUsingMark(primaryServerJNDICall);
+        System.out.println("errMsgs: " + trMsgs.toString());
+        assertFalse("Should find: " + primaryServerJNDICall, trMsgs.isEmpty());
+    }
+
+    /**
+     * Check if the returnPrimaryServer and primaryServerQueryInterval are working as designed. This test will shut
+     * down the primary server, wait for the time interval specified in the PrimaryServerQueryInterval, then turn the
+     * primary server back on
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    @ExpectedFFDC(value = "javax.naming.CommunicationException")
+    public void testNotReturnToPrimaryServer() throws Exception {
+        updateLibertyServer(false, 1);
+        String primaryServerJNDICall = "JNDI_CALL search\\(Name,String,SearchControls\\) \\[ldap://localhost:" + String.valueOf(ds2.getListenPort()) + "\\]";
+
+//        ServerConfiguration config = libertyServer.getServerConfiguration();
+//        ConfigElementList<LdapRegistry> ldaps = config.getLdapRegistries();
+//        LdapRegistry ldap = null;
+//        if (!ldaps.isEmpty() && ldaps.size() == 1) {
+//            ldap = ldaps.get(0);
+//        }
+
+        /* shutting the primary server down */
+        ds1.shutDown();
+        String returnUser = servlet.checkPassword(USER_DN, PWD);
+        assertNotNull("Should find user on checkPassword using " + USER_DN, returnUser);
+
+        /* restarting the primary server */
+        ds1.getLdapServer().startListening();
+        Thread.sleep(120000L);
+        libertyServer.setMarkToEndOfLog(libertyServer.getMostRecentTraceFile());
+        returnUser = servlet.checkPassword(USER_DN, PWD);
+        assertNotNull("Should find user on checkPassword using " + USER_DN, returnUser);
+        List<String> trMsgs = libertyServer.findStringsInLogsAndTraceUsingMark(primaryServerJNDICall);
+        System.out.println("errMsgs: " + trMsgs.toString());
+        assertFalse("Should find: " + primaryServerJNDICall, trMsgs.isEmpty());
+        updateLibertyServer(true, 1);
+    }
+
+    /**
+     * Check if the returnPrimaryServer and primaryServerQueryInterval are working as designed. This test will shut
+     * down the primary server, wait for the time interval specified in the PrimaryServerQueryInterval, then turn the
+     * primary server back on
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    @ExpectedFFDC(value = "javax.naming.CommunicationException")
+    public void testNotReturnEarly() throws Exception {
```

Comment:
Seems like this test could be combined with the nominal test. I know the issue said use the default of 15 minutes, but really, we could use the non-default and just call after sleeping 1/3 and 2/3 and verify we didn't switch. Then after the entire time is elapsed, verify we return.

Reply:
This would save a bunch of time in this test as well.